File: Calculator-ObjectiveC/Calculator-ObjectiveC/Calculator-ObjectiveC/ViewController.m
//
//  ViewController.m
//  Calculator-ObjectiveC
//
//  Created by Anthony Walker on 8/22/15.
//  Copyright (c) 2015 Anthony.Walker. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *displayLabel;

@end


@implementation ViewController

int firstNumber = 0;
int secondNumber = 0;
int runningTotal = 0;
int runningStringLength = 0;
NSString *operation = @"";
NSString *displayStack = @"";
NSString *runningString = @"";


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

//Appends the running string to whatever number is being pressed.
- (IBAction)numberPressed:(id)sender {
    NSString *buttonTitle = [(UIButton *) sender currentTitle];
    
    //Ensuring we don't have a leading zero
    if ([runningString isEqualToString:@"0"]){
        runningString = buttonTitle;
    }
    else{
        runningString = [runningString stringByAppendingString:buttonTitle];
    }
    
    _displayLabel.text = runningString;
}

//Depending on the operation, the overall expression is appended by the operation.
- (IBAction)operationPressed:(id)sender {
    NSString *buttonTitle = [(UIButton *) sender currentTitle];
    
    operation = buttonTitle;
    
    if ([operation isEqualToString:@"x"]){
        runningString = [runningString stringByAppendingString: @" "];
        runningString = [runningString stringByAppendingString: @"*"];
        runningString = [runningString stringByAppendingString: @" "];
    }
    else{
        runningString = [runningString stringByAppendingString: @" "];
        runningString = [runningString stringByAppendingString: operation];
        runningString = [runningString stringByAppendingString: @" "];
    }
    _displayLabel.text = runningString;
}

//Clears the fields and display
- (IBAction)clearButtonPressed:(id)sender {
    _displayLabel.text = @"0";
    runningString = @"";
}

//Evaluates whatever expression is currently in the running string and updates the value of the result
- (IBAction)equalbuttonPressed:(id)sender {
    //Saves application from crashing when user opens the app and just hits enter
    if ([runningString isEqualToString:@""]){
        runningString = @"0";
    }
    
    if (![runningString hasSuffix: @" + "]&&
       ![runningString hasSuffix: @" - "]&&
       ![runningString hasSuffix: @" * "]&&
       ![runningString hasSuffix: @" / "]){
        NSExpression *expression = [NSExpression expressionWithFormat: runningString];
        
        id result = [expression expressionValueWithObject:nil context:nil];
        
        _displayLabel.text = [NSString stringWithFormat:@"%@", result];
        runningString = [NSString stringWithFormat:@"%@", result];
    }

}

- (IBAction)deleteButton:(id)sender {
    runningStringLength = (int)[runningString length];
    //If the current expression is one digit, just set it to zero
    if (runningStringLength < 2){
        runningString = @"0";
    }
    //If the current expression ends with an operation, delete the operation
    else if ([runningString hasSuffix: @" + "]||
             [runningString hasSuffix: @" - "]||
             [runningString hasSuffix: @" * "]||
             [runningString hasSuffix: @" / "]){
        runningString = [runningString substringToIndex:runningStringLength - 3];
    }
    //Else, this is a multiple digit number, so just delete one number
    else{
        runningString = [runningString substringToIndex:runningStringLength - 1];
    }
    
    _displayLabel.text = runningString;
}

@end

File: Calculator-Swift/Calculator-Swift/ViewController.swift
//
//  ViewController.swift
//  Calculator-Swift
//
//  Created by Anthony Walker on 8/16/15.
//  Copyright (c) 2015 Anthony Walker. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var isTyping = false
    var firstNumber = 0
    var secondNumber = 0
    var operation = ""
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBOutlet weak var display: UILabel!

    //handles the pressing of a number
    @IBAction func numberPressed(sender: AnyObject) {
        var number = sender.currentTitle;
        
        if isTyping {
            display.text = display.text! + number!!
        } else {
            display.text = number
            isTyping = true
        }
    }

    //this occurs when the plus/minus +/- key is pressed
    @IBAction func calculationPressed(sender: AnyObject) {
        isTyping = false
        firstNumber = display.text!.toInt()!
        operation = sender.currentTitle!!
    }
    
    //this occurs when the equals = key is pressed
    @IBAction func equalsPressed(sender: AnyObject) {
        isTyping = false
        var result = 0
        secondNumber = display.text!.toInt()!
        
        if operation == "+" {
            result = firstNumber + secondNumber
        } else if operation == "-" {
            result = firstNumber - secondNumber
        }
        
        display.text = "\(result)"
    }
}

